pytest_rpc/helpers.py

```diff
@@ -216,10 +217,196 @@ def delete_instance(instance_name, run_on_host):
                                       action on.
 
     Raises:
-        AssertionError: If operation unsucessful.
+        AssertionError: If operation unsuccessful.
     """
     instance_id = get_id_by_name('server', instance_name, run_on_host)
     cmd = "{} openstack server delete {}'".format(utility_container,
                                                   instance_id)
+    run_on_host.run_expect([0], cmd)
+
+    assert (asset_not_in_the_list('server', instance_name, run_on_host))
+
+
+def create_instance(data, run_on_host):
+    """Create an instance from source (a glance image or a snapshot)
+
+    Args:
+        data (dict): a dictionary of data. A sample of data as below:
+                    data = {
+                        "instance_name": 'instance_name',
+                        "from_source": 'image',
+                        "source_name": 'image_name',
+                        "flavor": 'flavor',
+                        "network_name": 'network',
+                    }
+        run_on_host (testinfra.host.Host): A hostname where the command is being executed.
+
+    Example:
+    `openstack server create --image <image_id> flavor <flavor> --nic <net-id=network_id> server/instance_name`
+    `openstack server create --snapshot <snapshot_id> flavor <flavor> --nic <net-id=network_id> server/instance_name`
+    """
+    source_id = get_id_by_name(data['from_source'], data['source_name'], run_on_host)
+    network_id = get_id_by_name('network', data['network_name'], run_on_host)
+
+    cmd = "{} openstack server create --{} {} --flavor {} --nic net-id={} {}'"\
+        .format(utility_container, data['from_source'], source_id, data['flavor'], network_id, data['instance_name'])
+
+    run_on_host.run_expect([0], cmd)
+
+
+def verify_asset_in_list(service_type, service_name, expected_asset, run_on_host, retries=10):
```

Comment:
After playing with this for a bit, here are my suggestions for the `verify_asset_in_list`, `asset_is_in_the_list`, `asset_is_not_in_the_list` family of methods.

I keep confusing the base case and the positive case methods. Maybe they should be the same method. If the `expected_asset` defaulted to true there would be no need for the positive case.

It seems weird to be to have the `expected_case` change the return value behavior. I understand that by its name it is either producing the natural value or the inverted value of the `verify_asset_in_list` value, but I have been struggling with it. Would it be easier to understand if instead of the expected truthy value given as a parameter, this parameter indicated the precedence or the condition under which to double check the result?

Please take the following as a suggestion. I am including this only as a way of clarifying the issues mentioned above and showing my thoughts about how to express them in code.
```
def is_asset_in_list(service_type, service_name, run_on_host,
                     double_check_unless=True, retries=10):
    list = openstack_name_list(service_type, run_on_host)
    if retries < 1:
        if service_name in list:
            return True
        else:
            return False
    else:
        sleep(2)
        if (service_name in list) is double_check_unless:
            retries = 0
        else:
            retries = retries - 1
        return is_asset_in_list(service_type, service_name, run_on_host,
                                double_check_unless, retries)


def verify_asset_in_list(service_type, service_name, run_on_host):
    return is_asset_in_list(service_type, service_name, run_on_host)


def verify_asset_not_in_list(service_type, service_name, run_on_host):
    return not is_asset_in_list(service_type, service_name, run_on_host, False)
```